pkg/udevevent/monitor.go

```diff
@@ -147,22 +145,26 @@ func dispatchEvent(event UdevEvent) {
 }
 
 //Monitor start monitoring on udev source
-func Monitor() {
-	monitor, err := newMonitor()
-	if err != nil {
-		klog.Error(err)
-	}
-	defer monitor.free()
-	fd, err := monitor.setup()
-	if err != nil {
-		klog.Error(err)
-	}
-	for {
-		err := monitor.process(fd)
+func Monitor() <-chan error {
+	errChan := make(chan error)
+	go func() {
```

Comment:
Is this go routine created for processing errors only?

Reply:
Didnt understand this function @z0marlin. Too many channels now passing information.

Also, how many go routines (created from the code) will be present in NDM daemon now?

Reply:
The number of go routines hasn't changed. Its just that instead of starting the monitor as a go routine in udevprobe [here](https://github.com/openebs/node-disk-manager/pull/609/files/c7f7cab6a7f0a24c874b63d2f2ee684e3cc8e606#diff-7f7a5459cf891826a55f62700489f89dcd306dac941bf39d68c9091725b7456eL121), I'm starting it within the call to `Monitor()`. This allows me to return an error channel back to udevprobe.